File: app/src/main/java/com/starostinvlad/fan/Api/NetworkService.java
package com.starostinvlad.fan.Api;

import com.starostinvlad.fan.App;

import io.reactivex.schedulers.Schedulers;
import retrofit2.Retrofit;
import retrofit2.adapter.rxjava2.RxJava2CallAdapterFactory;
import retrofit2.converter.gson.GsonConverterFactory;

public class NetworkService {
    private static NetworkService mInstance;
    private Retrofit mRetrofit;

    private NetworkService() {

        RxJava2CallAdapterFactory rxAdapter = RxJava2CallAdapterFactory.createWithScheduler(Schedulers.io());


        mRetrofit = new Retrofit.Builder()
                .baseUrl(App.getInstance().getDomain())
                .client(App.getInstance().getOkHttpClient())
                .addCallAdapterFactory(rxAdapter)
                .addConverterFactory(GsonConverterFactory.create())
                .build();
    }

    public static NetworkService getInstance() {
        if (mInstance == null) {
            mInstance = new NetworkService();
        }
        return mInstance;
    }

    public FanserialApi getApi() {
        return mRetrofit.create(FanserialApi.class);
    }
}

File: app/src/main/java/com/starostinvlad/fan/Api/NetworkService.kt
package com.starostinvlad.fan.Api

import com.starostinvlad.fan.App
import io.reactivex.schedulers.Schedulers
import retrofit2.Retrofit
import retrofit2.adapter.rxjava2.RxJava2CallAdapterFactory
import retrofit2.converter.gson.GsonConverterFactory

class NetworkService private constructor() {
    private val mRetrofit: Retrofit
    val api: FanserialApi
        get() = mRetrofit.create(FanserialApi::class.java)

    companion object {
        private var mInstance: NetworkService? = null
        val instance: NetworkService?
            get() {
                if (mInstance == null) {
                    mInstance = NetworkService()
                }
                return mInstance
            }
    }

    init {
        val rxAdapter = RxJava2CallAdapterFactory.createWithScheduler(Schedulers.io())
        mRetrofit = Retrofit.Builder()
                .baseUrl(App.instance.domain)
                .client(App.instance.okHttpClient)
                .addCallAdapterFactory(rxAdapter)
                .addConverterFactory(GsonConverterFactory.create())
                .build()
    }
}